core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1425,7 +1428,8 @@ class ReplicaManager(val config: KafkaConfig,
                                entriesPerPartition: Map[TopicPartition, MemoryRecords],
                                requiredAcks: Short,
                                requestLocal: RequestLocal,
-                               verificationGuards: Map[TopicPartition, VerificationGuard]): Map[TopicPartition, LogAppendResult] = {
+                               verificationGuards: Map[TopicPartition, VerificationGuard]):
+  Map[TopicPartition, (LogAppendResult, Option[Uuid])] = {
```

Comment:
It seems cleaner to return Map[TopicIdPartition, LogAppendResult]?

Reply:
Hi @junrao, for a `TopicIdPartition`, topicId parameter in constructor([code reference](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/TopicIdPartition.java#L37)) has to be not null, which can happen in our case here. Hence, opted for `Map[TopicPartition, (LogAppendResult, Option[Uuid])]`. I was thinking of `Uuid.ZERO_UUID` for the null cases, but I was not sure.

Reply:
Perhaps we could create a new wrapper class TopicOptionalIdPartition?

Reply:
Hi @junrao , I have introduced a new wrapper class TopicOptionalIdPartition. Please review it when you can. Thanks!